agent/lib/kontena/workers/log_worker.rb

```diff
@@ -59,9 +61,23 @@ def start
         end
       end
 
+      async.watch_queue
       async.process_queue
     end
 
+    def watch_queue
+      loop do
+        sleep 10
+        if @queue.size > QUEUE_MAX_SIZE
+          warn "queue is full (size is #{@queue.size}), log lines are dropped until queue has free space"
+        elsif @queue.size > QUEUE_THROTTLE
```

Comment:
Make this something like `(QUEUE_MAX_SIZE * 0.8)`, it seems to always hover below the 1600 mark :)

```
W, [2017-04-07T14:29:02.735537 #1]  WARN -- Kontena::Workers::LogWorker: queue size is 1596
W, [2017-04-07T14:29:02.803540 #1]  WARN -- Kontena::Workers::LogWorker: queue size is 1596
W, [2017-04-07T14:29:12.766292 #1]  WARN -- Kontena::Workers::LogWorker: queue size is 1598
W, [2017-04-07T14:29:12.806389 #1]  WARN -- Kontena::Workers::LogWorker: queue size is 1600
```